src/framework/theme/components/cdk/overlay/overlay-trigger.ts

```diff
@@ -58,18 +64,21 @@ export class NbClickTriggerStrategy extends NbTriggerStrategy {
   // and then hidden right away
   protected click$: Observable<[boolean, Event]> = observableFromEvent<Event>(this.document, 'click')
     .pipe(
+      takeWhile(() => this.alive),
```

Comment:
I think your solution is too complicated. What do you think if we just `takeWhile` host element connected to the DOM?
```
takeWhile(() => this.host.isConnected)
```

I think, this solution will be enough for our requirements, also it'll be less code and much easier.